scripts/align_long_homopolymeres.py

```diff
@@ -0,0 +1,688 @@
+        # Handle gaps before this aligned segment
+        t_gap = t_start - last_target_end
+        q_gap = q_start - last_query_end
+
+        # Determine initial soft clipping for the query and adjust start_pos for target insertions
+        if first_time:
+            first_time = False
+            if t_gap > 0:
+                adjusted_start_pos = t_gap
+            if q_gap > 0:
+                add_op('S', q_gap)
+
+        # Adjust for gaps
+        # Compare t_gap and q_gap to determine order
+        elif t_gap > 0 and q_gap > 0:
+            # Both gaps exist, determine order
+            if t_start < q_start:
+                # Deletion occurs first
+                add_op('D', t_gap)
+                add_op('I', q_gap)
+            else:
+                # Insertion occurs first
+                add_op('I', q_gap)
+                add_op('D', t_gap)
+        else:
+            # Only one type of gap exists
+            if t_gap > 0: add_op('D', t_gap)
+            if q_gap > 0: add_op('I', q_gap)
+
+        # Add matched segment
+        segment_length = min(t_end - t_start, q_end - q_start)
+        add_op('M', segment_length)
+
+        # Update last processed positions
+        last_target_end, last_query_end = t_end, q_end
+
+    # Handle gaps after the last aligned segment - add soft-clipping for the remaining query
+    if last_query_end < seq2_len:
+        add_op('S', seq2_len - last_query_end)  # Treat remaining query as soft clipped
+
+    return adjusted_start_pos, ''.join(cigar), query_aligned[0][0], target_aligned[0][0]#, query_aligned[-1][1]
+
+def run_alignment_biopyhon(fa_seq, sequence, start_pos, sc_length, del_length, aligner):
+    # Perform the alignment between two sequences
+    if len(fa_seq) == 0 or len(sequence) == 0:
+        return 0, "", 0, 0, 0
+    for alignment in aligner.align(fa_seq, sequence ):
+
+        # Print each alignment's score and the alignment itself
+        start_pos_adjust, cigar, q_start, r_start = convert_aligned_to_cigar(alignment.aligned,  len(sequence))
+
+        start_pos = start_pos - sc_length - del_length + start_pos_adjust
+
+        #print(f"Cigar = {cigar}")
+        #print(alignment)
+        return alignment.score, cigar, start_pos, q_start, r_start
+    return 0, "", 0, 0, 0
+
+
+def reverse_complement(seq):
+    """Return the reverse complement of the given DNA sequence."""
+    complement_dict = {'A': 'T', 'T': 'A', 'C': 'G', 'G': 'C', 'N': 'N'}
+    # Generate the complement sequence
+    complement_seq = ''.join(complement_dict[nuc] for nuc in seq)
+    # Reverse the complement sequence
+    reverse_comp_seq = complement_seq[::-1]
+    return reverse_comp_seq
+
+
+
+def insert_operation_into_cigar(cigar, position, op_size, start_pos, op_type):
+    assert op_type in ['D', 'I'], "op_type must be 'D' for deletion or 'I' for insertion"
+
+    # Split the CIGAR string into its components
+    cigar_components = re.findall(r'\d+[MIDNSHP=X]', cigar)
```

Comment:
I think you have function for this

Reply:
I did not find any function like that, where can I find that?